crates/solver/src/liquidity/zeroex.rs

```diff
@@ -36,7 +36,7 @@ pub struct ZeroExLiquidity {
     pub orderbook_cache: Arc<OrderbookCache>,
 }
 
-type OrderBuckets = HashMap<(H160, H160), Vec<OrderRecord>>;
+type OrderBuckets<'a> = HashMap<(H160, H160), Vec<&'a OrderRecord>>;
```

Comment:
Why not to wrap it in an `Arc` like:

```rust
type OrderBuckets = HashMap<(H160, H160), Vec<Arc<OrderRecord>>>;
```

That way it is avoided the hard clones of the `OrderRecord`  here:
- [L203](https://github.com/cowprotocol/services/pull/2645/files#diff-9fc8dfc6f735cce917d647b7f913cc65d3d314a76f8a7e9f31214cabc661518cR203)
- [L206](https://github.com/cowprotocol/services/pull/2645/files#diff-9fc8dfc6f735cce917d647b7f913cc65d3d314a76f8a7e9f31214cabc661518cR206)

The signature of the function could change to;
```rust
fn get_useful_orders(order_buckets: OrderBuckets, orders_per_type: usize) -> Vec<Arc<OrderRecord>> {
```

That way we avoid making any copy of `OrderRecord` but always pointing to the reference in memory :thinking: (In case we want to fully avoid making hard cloning)

Reply:
Yeah, brought this up already with @squadgazzz and he's currently working on that. Actually he came up with some other improvement which will make this PR obsolete so it can be closed.